framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -148,31 +143,134 @@ private static Pattern anchored(String regex) {
             return parseArray(expression, context, path);
         } else if (isMethodCall(expression, context)) {
             return parseMethodCall(expression, context, path, env);
-        } else if (isMemberSelect(expression, context)) {
-            return parseMemberSelect(expression, env, context, path);
+        } else if (isMemberSelectOrMethodChain(expression, context)) {
+            return parseMemberSelectOrChainedMethodCalls(expression, env, context, path);
         } else if (isParentheses(expression, context)) {
             return parseParentheses(expression, context, path);
         } else {
             throw constructParserException(expression);
         }
     }
 
-    private static boolean isMemberSelect(String s, FlowExpressionContext context) {
-        return parseMemberSelect(s) != null;
+    /** Return true iff s contains a dot (.) and a method call. */
+    private static boolean isMemberSelectOrMethodChain(String s, FlowExpressionContext context) {
+        return parseMemberSelectOrChainedMethodCalls(s, context) != null;
     }
 
     /**
-     * Matches a field access. First of returned pair is object and second is field.
+     * Matches expressions containing a dot (.) and a method call. If expression contains a method
+     * call whose caller is a method call as well (method chain), first of the returned pair is the
```

Comment:
Should "caller" be "receiver" here?  I don't see how this has access to the caller.

Reply:
If the expression contains are multiple method calls, which one is chosen?  This comment does not make it clear.
Is the last top-level method call in the expression the one chosen, or something else?